tests/CallerTests.cs

```diff
@@ -47,6 +47,30 @@ public void ItCanGetMemoryFromCaller()
         callback.Invoke();
     }
 
+    [Fact]
+    public void ItCanGetCachedMemoryFromCaller()
+    {
+        var memories = new HashSet<Memory>();
```

Comment:
Nit: To guarantee that the `HashSet` checks for object reference equality only, even if `Memory` would override `Equals`/`GetHashCode` in the future, you can pass `ReferenceEqualityComparer.Instance` in the constructor.

Reply:
Fixed